### PR TITLE
Python: Proper separation of unit and integration tests

### PR DESCRIPTION
**Pull request by awharrison-28**
### Motivation and Context
This PR introduces the ability to run formal integration tests for python using pytest - separated from unit tests for easier automation.

### Description
- moved `python/tests/end-to-end` directory to `python/tests/integration`
- renamed integration tests to test_* pattern so pytest will detect them
- import `GenerationConfig` for hugging face completions - according to logs, this is the recommended practice

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by shawncal — DISMISSED** [review]

File: .github/workflows/python-integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#
+# This workflow will run all python integrations tests.
+#
+
+name: python-integration-tests
+
+on:
```
Comment (lemillermicrosoft) [inline .github/workflows/python-integration-tests.yml:7]: nit: i think it'd be good to be consistent between the two. I think on push makes most sense, do we think the schedule will help identify issues?

Comment (lemillermicrosoft) [reply .github/workflows/python-integration-tests.yml:7]: I could see it help identify breaking requirements changes from external packages. I'd prefer locking them down though for that reason :)

Comment (awharrison-28) [reply .github/workflows/python-integration-tests.yml:7]: Removing this file to PR separately. Will pick up this conversation there.

**Review by alexchaomander — APPROVED** [review]
LGTM!

**Review by mkarle — APPROVED** [review]

**Pull request merged**